CSharp/ArticleProjects/DotNetTutorial/Program.cs

```diff
@@ -286,17 +286,37 @@ private static string GetContainerSasUrl(CloudBlobClient blobClient, string cont
         /// <param name="resourceFiles">A collection of <see cref="ResourceFile"/> objects representing blobs within
         /// a Storage account container. The StartTask will download these files from Storage prior to execution.</param>
         /// <returns>A <see cref="System.Threading.Tasks.Task"/> object that represents the asynchronous operation.</returns>
-        private static async Task CreatePoolAsync(BatchClient batchClient, string poolId, IList<ResourceFile> resourceFiles)
+        private static async Task CreatePoolIfNotExistAsync(BatchClient batchClient, string poolId, IList<ResourceFile> resourceFiles)
         {
-            Console.WriteLine("Creating pool [{0}]...", poolId);
-
-            // Create the unbound pool. Until we call CloudPool.Commit() or CommitAsync(), no pool is actually created in the
-            // Batch service. This CloudPool instance is therefore considered "unbound," and we can modify its properties.
-            CloudPool pool = batchClient.PoolOperations.CreatePool(
-                poolId: poolId,
-                targetDedicated: 3,                                                         // 3 compute nodes
-                virtualMachineSize: "small",                                                // single-core, 1.75 GB memory, 225 GB disk
-                cloudServiceConfiguration: new CloudServiceConfiguration(osFamily: "4"));   // Windows Server 2012 R2
+            CloudPool pool = null;
+            try
+            {
+                pool = await batchClient.PoolOperations.GetPoolAsync(poolId);

```

Comment:
I believe @darylmsft is right -- the code as it sits currently seems incorrect (GetPool will never throw a PoolExists error). Why does this method have different semantics than `CreatePoolIfNotExistAsync` in GettingStartedCommon? It seems like you wanted something very similar to what `CreatePoolIfNotExistAsync` does in GettingStartedCommon -- which I think is totally reasonable to put here, but why not just use a snippet of that logic here like I suggested above (what I mean is, instead of `GetPool` followed by `Commit`, just do a `Commit` and catch the `PoolExists` exception if it's thrown).